### ISSUE TITLE
Website redesign proposal

### ISSUE THREAD

**Issue by birjolaxew** [simple-icons/simple-icons#980]
The current design has a few issues:
- Difficult to see some icons/text (e.g. JavaScript)
- Have to style dark/light icons differently to mitigate this - still not perfect

A few design proposals:
- [Variant proposed by @abenjamin765](https://user-images.githubusercontent.com/3742559/45262339-90ba9b80-b41d-11e8-883f-85c9eed1bd77.png) in #974 (see that PR for code)

- [Variant with drop shadows behind icons](https://user-images.githubusercontent.com/4542461/45362191-78d94800-b5d4-11e8-8b48-1d6a07fd32ab.png)

- [Variant with separated title + subtitle](https://user-images.githubusercontent.com/4542461/45362525-57c52700-b5d5-11e8-89cf-3b56a875fbc8.png)

- [Variant with separated title](https://user-images.githubusercontent.com/4542461/45362604-8cd17980-b5d5-11e8-9e7f-52fa26ffefd8.png)

This issue is for discussing whether we should implement a different website design - and if so, which one.

**Comment by birjolaxew** [comment]
I am personally not entirely happy with any of the proposals, and prefer the current design (although that might just be me being used to it). Other design proposals are *very* welcome.

**Comment by abenjamin765** [comment]
Some things we could explore:
- Since the brand colors are usually applied to the logo, should we explore a treatment that showcases them that way?
- Are there any thoughts about the size of the icons displayed? 
- How can we best care for color contrast to be inclusive of those with low vision or colorblindness?
- Should we make more affordance for the download action? 
- How do you currently gather feedback from users outside of github? 
- What else could we improve?

**Comment by ericcornelissen** [comment]
I do think it is worth discussing a redesign, with the main issue of the current design being color contrast/visibility of some of the text. Also with issues like #755 in mind (and the affordance of the download action, as pointed out by @abenjamin765) it will be useful to redesign the icon blocks.

Of the proposed designs so far I like the [Variant with separated title + subtitle](https://user-images.githubusercontent.com/4542461/45362525-57c52700-b5d5-11e8-89cf-3b56a875fbc8.png). Either way I think using drop shadow to get rid of the dark/light icons problem is a good start, as I think we should keep a colored background behind the SVG.

To answer some of the questions by @abenjamin765:
- _Since the brand colors are usually applied to the logo, should we explore a treatment that showcases them that way?_ Actually, the SVGs do not inherently have any color, and displaying them with a color may be misleading I think.
- _How do you currently gather feedback from users outside of github?_ So far not at all, and I'm not sure if we should focus on that now too much. Also, I wonder how many users we have that don't know about GitHub at all 🤔 
- _What else could we improve?_ I think that that is what this issue is for

**Comment by ericcornelissen** [comment]
That is a good list to start from @birjolaxew 👍 I want to point out #743 and the [simple-icons-pdf](https://github.com/simple-icons/simple-icons-pdf) project with regards to the explicit download action as well.


> Most of these would probably be best handled by having a modal opened when selecting an icon (or expanding a section below the icon), similar to the previous example of https://materialdesignicons.com/

I think we should avoid having a modal (at least in the classic sense of the word, e.g. [bootstrap modals](https://getbootstrap.com/docs/4.1/components/modal/)) as this may break the user flow for our purposes.

**Comment by abenjamin765** [comment]
I'm not usually a fan of horizontal scrollers... but this could be an interesting way to showcase brands with the right controls (click and drag or keyboard scrolling). 

![image](https://user-images.githubusercontent.com/8422137/45859853-235e1300-bd19-11e8-9634-7243fcee339f.png)

**Comment by xdesro** [comment]
Hey guys! Happy Hacktoberfest. Haven't committed in a while but hopefully y'all will have me back ❤ I'm working on a design for this right now -- sort of trying out this card concept, I'd love to hear your thoughts.
- Each card still contains the name of the brand and primary hex color. Instead of the hex color as just a string, it is featured as the text on a themed chip. The text color of the hex color (oh boy) will either be black or white and use some WCAG contrast-checking library to decide which is correct.
- There is now an explicit link to the brand guidelines.
- The lower portion of the card is designated for downloads. Currently, I envision it as clicking the black punched out icon will download the SVG, but clicking different formats (if we choose to go that route) will download that format. In this section I could also see a "copy SVG code" or "copy JSON" option being useful as well. 

Again, stoked to hear what you think. Disclaimer: I whipped this together pretty quick, so it's not too polished or opinionated, just trying to communicate the concept! I'd really love to work it into something we can all like. 

![screenshot-1](https://user-images.githubusercontent.com/13542610/46477742-956a3980-c7a8-11e8-8139-bf6139ba0f41.png)

**Comment by birjolaxew** [comment]
@xdesro My personal thoughts:

- I really like the design! It lets us expand with more information easily, and looks clean and modern.
- The new design removes the focus from the icons, and places it on the brands instead. This could make it significantly harder to scan a page full of icons quickly.

I'd really like to see how it looks on a page full of icons, as the most important part of the redesign would be making it easy to see and scan for the icon you want.

**Comment by xdesro** [comment]
@birjolaxew This is really good feedback. I've got some ideas of my own, as well. Let me get back to you later today with an updated concept and a full-page mock.

**Comment by ericcornelissen** [comment]
> The new design removes the focus from the icons, and places it on the brands instead. This could make it significantly harder to scan a page full of icons quickly.

First off, I like your design idea as well @xdesro 😃 There are three major problems I have with your approach, and both can be solved by focusing on this issue pointed out by @birjolaxew.

- My first thing is with regards to the duplication of the icons (once shown partially at the top of the card and then in full at the bottom. I think we should stick to each icon appearing only once, keeping one large version of the icon (probably at the top of a card). 
- Secondly only partially showing the icon can easily lead to complications. Your JavaScript example actually illustrates this pretty well. It cuts of the most important/iconic part of the icon, namely the letters "JS". Truth be told in this case you can still see the "J", but you might imagine some icons for which this might cause problems. However, I do like the overlap of the brand information on top of the icon, so you might keep that. Perhaps there could be some hover animation that shows the icon in without overlap?
- The download actions are not clear enough in my opinion. Clicking the small icon to download the SVG is not clear at all, and the other buttons to download the icon in different formats are just too small.

**Comment by ericcornelissen** [comment]
First of all @xdesro, great to see the effort you're putting into this 😮 🎉 

I like what you did with the `Comfortable vs Compact` design , and I really like the compact design! Two issues I see with it are:

- It will be harder to fit the buttons for all icon types, which I think we **should** still provide in compact mode.
- Displaying the color code vertically might cause some confusing with regards to copying the value, which is its main purpose I think, otherwise we could get away with just having the colored sidebar for the card. So I think this should be changed somehow, but I don't see any obvious solutions.


> Firstly, I think that the search navigation toggles (sorting alphabet/color and now comfortable/compact) should persist for users:

I totally agree, I think this is just a quality of life feature we haven't though of yet. Just an FYI, I think we should separate this idea from this thread, so I opened #999 for that (haven't included layout preferences yet as it is not yet part of the site).

> Second, when alphabetical sorting is active, I've added an alphabet scroll element on the left side to clearly indicate how far you've scrolled through the many, many icons in SimpleIcons. Once you've scrolled past the start of the icons, it fixes itself on the page and animates stepping through each letter. I also feel the sorting/filtering nav should also become sticky.

Another quality of life feature that I'd love to see 😄 


> I think that the duplication v. non-duplication conversation is mostly aesthetic, but I definitely see where you're coming from with the cropping feeling unnecessary...

One thing I didn't mention in my original comment but is a related concern is that displaying the icon twice might impact performance as all icons (almost 500 by this point) have to be rendered twice. Although to be fair, we might think about dynamically loading icons as well 🤔 


* * *


@birjolaxew, I agree with everything you said expect the following points

- _Not particularly fond of the new `<h1>` font - but that is very much so personal opinion._ -- but as you said, it is very much personal. To be fair I would be okay with a size with would fit one line (on desktops) as well
- _Also not a fan of the header (SimpleIcons title and GH/NPM/about links) breaking the horizontal width - of the rest of the page._ -- I see your point, but I think having them in line with the rest of the page won't look good either...
- _The alphabet indicator seems a bit redundant; would probably compete a bit with the search function._ -- It might be a bit feature creeping, but I don't think it really "competes" with search rather then being a replacement. But sure, this should be implemented as a separate feature from the redesign.

**Comment by xdesro** [comment]
Hey gang,

As I understand it, here are the actionables from this review: 
- Decrease negative space in comfortable layout. 
- Increase font sizes to a minimum of 12px on both layouts
- Minor adjustments to sizing on compact card to balance header and icon sizing
- Add hover state to compact view cards to show extended options. As always, clicking the icon downloads the icon. 
- Removed secondary font. Looks just as good solely Roboto Mono, and feels more simple which I like.
- Reduced header width to match content container, so it doesn’t fill the full width (I personally think it's better the other way, but that's just personal preference!)
- Removed alphabet scroller. It’ll be a cool enhancement down the road but let’s not get hung up on it now!

## Updated Mocks:

#### Comfortable Cards
![comfortable cards](https://user-images.githubusercontent.com/13542610/46505694-1e0bc880-c7ef-11e8-8fe7-9c6daa2e92d0.png)

#### Compact Cards
![compact cards](https://user-images.githubusercontent.com/13542610/46505707-27953080-c7ef-11e8-9611-3208ea8f3a63.png)

#### Site Mocks
![desktop hd comfortable](https://user-images.githubusercontent.com/13542610/46505714-311e9880-c7ef-11e8-898f-9e6bb13f7d03.png)
<hr />

![desktop hd compact](https://user-images.githubusercontent.com/13542610/46505715-311e9880-c7ef-11e8-99a7-d2d61e0a9bc5.png)


## Thoughts: 
> The search / sort toggles / layout toggles look a bit out of place in the new design

Ideally, I’d like to do some new iconography for this to make it feel more cohesive, but if I’m completely honest, I think it doesn’t look too bad!

>Displaying the color code vertically

To solve this, I either see a hover state on the color tab that swaps the “#000” text for a clip board so it’s very obvious, OR adding a “copy hex code” entry to the “view more” state of the compact card.

> Displaying the icon twice might impact performance 

This definitely resonates with me. Would you be open to pursuing dynamic/progressive loading? My vote is to implement as mocked and take out the secondary icon if performance feels like it suffers?

Thanks for all your help!

**Comment by birjolaxew** [comment]
@xdesro After seeing the header at the same width as the content, I agree with you both that it's better as full width.

Other than that, I really like the newest iteration! I don't really have anything else to add, it looks great to me (with full width header ;) )

**Comment by ericcornelissen** [comment]
> To solve this, I either see a hover state on the color tab that swaps the “#000” text for a clipboard icon and cursor change so it’s very obvious ...

I like that idea, there is an argument to be made that the action should be clear even without hover. On the other hand, the compact mode is more for power users anyway so that perhaps isn't as much of a problem. That being said:

> ... adding a “copy hex code” entry to the “view more” state of the compact card.

I personally dislike this idea, but I'm completely open to implementing it anyway. Partly because of my argument that the action should be clear without hover.


> Would you be open to pursuing dynamic/progressive loading? My vote is to implement as mocked and take out the secondary icon if performance feels like it suffers?

Sounds like a plan :+1: 

* * *

All in all your design, @xdesro, looks very modern and cool. And it also includes lots of space for features that are not yet implemented (e.g. brand guidelines #755). So good work :slightly_smiling_face:

**Comment by xdesro** [comment]
Feels like we're getting close! If it's cool with y'all, I'll work on getting a PR going for this over the weekend -- then we can discuss implementation specifics more in the comments of that. Thoughts?

**Comment by birjolaxew** [comment]
@xdesro Sounds great to me. Note that we're looking to move away from a Jekyll-based website and over to one that's built however we want (webpack? custom Node script?), which will then be ran on every release by our CI (#898). It'd be great if that could be included in this rewrite. If you'd rather not take on that extra work, I could create a new branch with whichever build tool you want that you can base off of.

**Comment by xdesro** [comment]
@birjolaxew I'm willing to give it a shot! If you've got particular specifications I'd love the challenge, otherwise I'll just go forth with static-gen and webpack in mind?

**Comment by abenjamin765** [comment]
Really nice work, gang! Love the compact idea. The vertical hex code seems hard to read, but I dig the focus on the icon and making the metadata visible.

**Comment by abenjamin765** [comment]
Here's my swing at a card design: 
![image](https://user-images.githubusercontent.com/8422137/46589543-acdf4600-ca78-11e8-997e-732efe7ab0f3.png)

**Comment by abenjamin765** [comment]
Another... 
![image](https://user-images.githubusercontent.com/8422137/46589683-8a4e2c80-ca7a-11e8-8759-168cd050fa66.png)

**Comment by ericcornelissen** [comment]
Thank you for your input on this as well @abenjamin765. I think your design might look better for colored ordering, but as pointed out by @xdesro "... typically alphabetical sorting is more useful." So I'm ignoring that for now.

Furthermore, even though I like @xdesro's design better overall, I'd like to suggest to him to try and incorporate the hex-value in the compact cards a similar manner to your suggestion @abenjamin765 (i.e. keep the color on the side but display the hex-value itself horizontally, possible in the `view more` section).

Also, I like what you did with the different background colors in the larger cards @abenjamin765. @xdesro I invite you to experiment with this style as well (for more references, have a look at [this tweet](https://twitter.com/steveschoger/status/897849211110273024)).

**Comment by abenjamin765** [comment]
I agree with alphabetical being a better way of organizing things. My mockup was only meant to show the card design :)

**Comment by xdesro** [comment]
@abenjamin765 I think you're right about the sideways hex thing. I'm gonna go ahead and include thex on the front side of the compact card between the icon name and the "view more" button, maybe with a clipboard icon. Closer to have you have it! 

@ericcornelissen @birjolaxew  I'm about 30% of the way done building the facelift out on my fork of the project. I'll link when I have something to show!

**Comment by abenjamin765** [comment]
@xdesro - What's the purpose of having a toggle between compact and a block view? If compact has enough information on it to be useful, it's not super clear what the block version offers. Might be simpler to only have one view...though we may have a case for a list/table view if there are users that need to focus more on the metadata. 

In your compact design, you should be able to get away with swapping the "view more" link with the hex value and extending the svg/png download buttons to the right similar to what I was exploring. 

Nice work so far!

**Comment by ericcornelissen** [comment]
@abenjamin765 

> What's the purpose of having a toggle between compact and a block view?

The advantage may be that the block view is just a bit easier on the eyes, some apps (and many Google apps) actually give you this option with their layout. But to be fair, it may not be the most important thing for now...


> In your compact design, you should be able to get away with swapping the "view more" link with the hex value and extending the svg/png download buttons to the right similar to what I was exploring.

I do agree that it would be nice to get rid of the "view more" button, but I'm not sure if pushing the buttons to the right (or left for that matter) will work in @xdesro's design.

**Comment by moldypenguins** [comment]
> 
> 
> @abenjamin765
> 
> > What's the purpose of having a toggle between compact and a block view?
> 
> The advantage may be that the block view is just a bit easier on the eyes, some apps (and many Google apps) actually give you this option with their layout. But to be fair, it may not be the most important thing for now...
> 
> > In your compact design, you should be able to get away with swapping the "view more" link with the hex value and extending the svg/png download buttons to the right similar to what I was exploring.
> 
> I do agree that it would be nice to get rid of the "view more" button, but I'm not sure if pushing the buttons to the right (or left for that matter) will work in @xdesro's design.

perhaps flipping the card on hover to show the "back" instead of having a view more button?

**Comment by metaa** [comment]
I could drop in here, I'm a senior web developer.

---

From the first post I really like [the example](https://user-images.githubusercontent.com/5056880/54883180-64b7b280-4e63-11e9-9aef-bb0fad6c3722.png) that has both a shadow and a stronger outline.

---

@abenjamin765:
> Since the brand colors are usually applied to the logo, should we explore a treatment that showcases them that way?

Yes that's something I thought too, this would actually be a lot better than them all being black in my opinion.

The only thing to note is, that very bright icons would then need a black card with inverted text, so it's all visible still.
I would recommend using the `score` function from [this NPM package](https://www.npmjs.com/package/wcag-color) to determine the best contrast for a color. I've used it in the past and it's really great.

**Example:**
```js
const wcag = require('wcag-color')

let logoColor = '#ABC'

if(wcag.score(logoColor, '#FFF').startsWith('AA')) {
    // good with white background
} else {
    // good with black background
}
```

---

[This design](https://user-images.githubusercontent.com/13542610/46477742-956a3980-c7a8-11e8-8139-bf6139ba0f41.png) by @xdesro isn't bad either but I would fear it would get a little too big with all the icons on one page (and a pagination doesn't sound like a good solution either)

What I like about [this one](https://user-images.githubusercontent.com/13542610/46497909-1a6d4700-c7d9-11e8-96d8-cb9e7b8c08ad.png) is the letter navigation on the side. Great idea.

---

@moldypenguins:
> perhaps flipping the card on hover to show the "back" instead of having a view more button?

Personally I really like this idea, though we should consider if this wouldn't get a bit annoying when looking at multiple icons 🤔.
Flip on click or hover?

---

If I find some time, I could try and combine some of these ideas, maybe we'll find something everone would be happy with 🙂

**Comment by ericcornelissen** [comment]
With regards to [this design](https://user-images.githubusercontent.com/5056880/54883180-64b7b280-4e63-11e9-9aef-bb0fad6c3722.png), [the design by @xdesro](https://user-images.githubusercontent.com/13542610/46477742-956a3980-c7a8-11e8-8139-bf6139ba0f41.png) and the "flip" idea (by @moldypenguins) I would say that the main advantage of the design by @xdesro is that the all information and actions are available at once. The drawback of "flip" on hover is that it requires interaction, which might not occur for a myriad of reasons. "flip" on click is essentially the same as a "view more" button, but that would be a cool animation if we decide to go with some on click event.

Either way, I would prefer the new design to make it easy to download different versions of the logo (svg, pdf, png?) an get (or at least see) the color hex **in a single click**. Ideally it is also possible to access brand guidelines in a single click.

As for coloring the icon itself, I'm personally against that because of the "very bright icons" problem (also, I usually don't display the icons with their brand color by default). Not only does one need to implement some kind of logic to find such icons, you also need to adept the design of the website to make this look good... In that regard I'm a huge fan of the designs by @xdesro and @abenjamin765 


> If I find some time, I could try and combine some of these ideas, maybe we'll find something everone would be happy with 🙂

I wanted to get working on that in the near future as well, but go ahead if you want 👍

**Comment by el1f** [comment]
Hello everyone, I got interested in this issue and tried my hand at a redesign for the website. Please let me know what you think about it. I'd love to actually work on the implementation of a redesign so I'd love to get as much input as possible.

**Idle home page**
![idle](https://user-images.githubusercontent.com/10723208/67800596-d1dd9700-fa87-11e9-8127-88037a0e7403.png)
I love how the current implementation shows a lot of icons on-screen so one of the main focuses of my main view was to try and keep this aspect by keeping the icon card sizes small. I took a lot of inspiration from @xdesro's mockup and sites like fontawesome and noun project when it comes to showcasing the icons. Ideally, I thought about having a second layout that hides the action buttons on the cards and shows them on hover. We could also play around with other ideas for alternative layouts.
Since I think that most people would be fairly coherent in the format they require when they download multiple icons during a single session, I'd have it as an option on the sidebar that would allow to pick it once and just press on the download button on the card to get the file as the predefined format.
The action buttons lack a label, but I think an on-hover tooltip would be an honest compromise since we wouldn't offer that many actions and they are fairly straightforward. I used generic Material icons here but we could make custom icons to make the function of the button more understandable.

**Hover modifier**
![hover](https://user-images.githubusercontent.com/10723208/67801323-39e0ad00-fa89-11e9-8d82-18adb42f3292.png)
On hover, I'd color the icon with the color of the brand. There is no showcase on it on the mockup but when it comes to white or light icons, we could have a drop-shadow on the icon to make it easily distinguishable from the card.
I'd also explore the possibility of having the background of the card change, to give the white interface a well-deserved dash of color.

**Icon detail view**
![active](https://user-images.githubusercontent.com/10723208/67801718-12d6ab00-fa8a-11e9-8422-fb5790561de9.png)
I love the idea of being able to do some light editing on the website itself like setting the fill color or setting a stroke width. This would also give us an outlet to add more features in the future. We could also have some `pre` blocks with the code for the most relevant versions (pure SVG, React, etc.).

**Kits**
![collection](https://user-images.githubusercontent.com/10723208/67801856-5c26fa80-fa8a-11e9-80c3-1e0a5ae65bbd.png)
Instead of downloading the icons one-by-one we could add a collection feature, powered by local storage, which could allow the user to pick several icons and download them all in a single `.zip` file.
It could also allow for the application of a global modifier for the whole collection before downloading (change the color of all the icons before downloading and so on).

Awaiting your comments on this proposal guys! Can't wait for them :)

**Comment by foo-dogsquared** [comment]
Wow! Nice work, @el1f ! 

Since I'm not one of the original maintainers, I can only infer from the discussion on what is the main objective/use cases of the redesign that they were going for. 
Thus, I don't think my opinion bears weight as much as the original maintainers (i.e., @ericcornelissen, @birjolaxew). 

With that out of the way, it's time to give some feedback. :)

First and foremost, I'm in favor of how the layout looks. 
It presents all of the primary user info (i.e., hex color, icon itself) and actions (i.e., downloading the SVG, copying the SVG code) in a pretty neat card. 
Through out of the icon card layouts that have been proposed here, yours is my favorite so far because of its simplicity and neatness — not too spacey and not too compact. 
The only gripe with it is the action buttons look small though I'll hold that statement back until I see the implementation. 

I also like the hover modifier. 
I think it's a harmless addition and can be pretty helpful. 

Regarding with the more advanced options like the light editing options, I'm leaning to wonder if it's an unnecessary feature. 
Though, I'm still open on the idea. 
I like how the layout can be easily distinguished and tucked behind a click. 

Moving on to the next part, I really like the Kits feature (though, it might just be because of my bias related to what I'm about to say). 
One of my main gripes with the current website is the lack of option for selecting multiple icons at once. 
I eventually created a [website](https://simple-icons-pwa.netlify.com/) just for that since I often find myself downloading a set of social media icons for different events (plus it's handy). 
The lack of selecting multiple icons feels lacking for a website that showcases an icon set (at least to me). 

I feel like the placement of the kits feature is great and it doesn't get much in the way of using the website. 
The only thing missing with it (from what I can tell from the images anyway) is some convenience functions like clearing all icons from the selection or selecting all of the current icons in the selection grid. 

Overall, I think the proposed website layout is simple and familiar to use given that the site offers more features than ever.

**Comment by birjolaxew** [comment]
Wow, thanks a lot for the concept @el1f -you've obviously put a great deal of thought into it. I'll be breaking up my feedback in two sections.

**TL;DR:** Really dig the layout, questioning the need for some of the functionality, think the visual design is a bit too neutral/bland.

## Functionality and layout

I dig the horizontal layout with both sidebars. That would allow us to display a lot more metadata about the individual icon (if we wish to add more in the future), while also giving a better look at the icon. I am a bit worried about how it would play out on phones, tablets or small laptops; I assume the left sidebar would become horizontal on top of the icons, with the right sidebar either disappearing entirely or becoming a modal/slidein?

As for the functionality, I initially really liked most of it. As I spent more time thinking about it, I did start to wonder whether most of it was necesarry tho: it is rare for people to download more than a couple of icons from our site at a time, and at that scale I think the browser's built-in download functionality is fine for grouping them. I would also question whether the ability to download as raster would be needed - I can't really think of an application for our icons that an SVG wouldn't do the job of (and spreading the gospel of the vector god is always good ;) ). The stroke width doesn't really make much sense, so I'll assume that was added mostly to get to 4 options ;)  
I think that in general we would prefer keeping the user experience as simple as possible, unless the added functionality is really needed.

## Design

My general impression is that the visual design is pretty bland, with little contrast to help focus the eye on important parts. This is very obvious on e.g. the dropdowns, where both the contrast between text and dropdown, and between the dropdown and background is poor - but I think it is a general problem with the colors chosen. No part really stands out, so no part is given special focus. Some of the previous proposals show other examples of this (e.g. @abenjamin765's [second card design](https://github.com/simple-icons/simple-icons/issues/980#issuecomment-427705290) has better contrast than his first, and @xdesro's [compact cards](https://github.com/simple-icons/simple-icons/issues/980#issuecomment-427185048) have significantly better focus on the icon than his comfortable ones).

That's the end of my more or less objective feedback. Some personal biases: I really like how the colors look on the current website, which is partially kept by @xdesro's and @abenjamin765's card designs, while this proposal almost removes them entirely. I'm also not a fan of the very rounded corners, but that's entirely personal preference.

## Conclusion

In general, I *really* like what you've done with the layout. I do think that some of the suggested features will be so rarely used that they mostly add complexity (although they are obviously the most fun to implement ;) ), but the layout with two sidebars and the icons in the middle is something I think that we want to move towards when/if we update the website. The visual design will need some rework though.

**Comment by el1f** [comment]
I ended up going ahead and coding the first version due to boredom today. Now that I read your feedback I'll try to introduce some changes to find a design that better suits all of your suggestions.
Thank you for the time you put unto your feedback, it was extremely appreciated :)

In the meantime you can play around with a terrible first version of the website 😂
http://el1f.github.io/simple-icons-website

**Comment by ericcornelissen** [comment]
Decided to try my hand and building a redesigned website based on some of the submissions here, see the result at https://ericcornelissen.github.io/simple-icons-website/ (and source at [ericcornelissen/simple-icons-website](https://github.com/ericcornelissen/simple-icons-website)). As of writing this there are still definitely some things missing, but I think we may start asking visitors of https://simpleicons.org/ to give it a try after a few more things are fixed.

**Recognition**: What I ended up with is mostly inspired by [@xdesro's mockups](https://github.com/simple-icons/simple-icons/issues/980#issuecomment-427185048), but I took some inspiration from other's as well. though I think it would be great if some of the functionality of [@el1f's design](https://github.com/simple-icons/simple-icons/issues/980#issuecomment-547591284) can be inlcuded as well.

**Comment by fbernhart** [comment]
Here is some feedback from my side:

### Already great:
- I like the "card concept" with all the necessary information and buttons available without any additional clicking or hovering. Great work!
- The "copy to clipbord" functionality for the colors and SVG on hover is a great idea!
- Overall the layout is very tidied up and simple.

### Could be improved in my opinion:
- My overall concern is the website loading speed. I'm not sure how you've done it, but your version is already loading faster than our current website. So kudos for that! But I think it's still pretty slow and should be improved (if possible).
Could we implement some kind of lazy loading? So the website is loading the first 100 (or 200, or 300, ...) icons and then if you scroll down (or click on a "Next Page" button), it will load the next 100? So we don't have to load all of the 1500+ SVGs at once.

- I'm not too sure if I'm a big fan of the current way of displaying the same logo twice (once in black and once in light grey and half cut off). Either I would prefer to display the logo in light grey without cutting it off:

![image](https://user-images.githubusercontent.com/70264417/103409648-ffc8da80-4b67-11eb-9884-fbd0f75b004f.png)

Or just displaying it once in black. Following are just some suggestions for the card layouts with the icon displayed only once:

1. ![Untitled_1](https://user-images.githubusercontent.com/70264417/103409673-18d18b80-4b68-11eb-9d01-cf0fc1fdff5d.png)
2. ![Untitled_2](https://user-images.githubusercontent.com/70264417/103409675-1cfda900-4b68-11eb-84b8-8069a9c89711.png)
3. ![Untitled_5](https://user-images.githubusercontent.com/70264417/103409734-65b56200-4b68-11eb-8f32-25f7c0d2a19f.png)
4. ![Untitled_4](https://user-images.githubusercontent.com/70264417/103409680-20913000-4b68-11eb-923b-bbc238d5aafc.png)
5. ![Untitled_6](https://user-images.githubusercontent.com/70264417/103409865-0572f000-4b69-11eb-847b-685ee8111aeb.png)

I guess by displaying (and therefore loading) the SVGs only once, we could as well improve the website loading speed significantly.

 - Another last suggestion: I think it would look better, if the cards in the last row would keep their size. Currently they are stretched out:

![image](https://user-images.githubusercontent.com/70264417/103352876-b6b15180-4aa7-11eb-8b7c-772eb636616f.png)
 
 
 
 Once again: Amazing work so far @ericcornelissen! 🎉 🏆

**Comment by ericcornelissen** [comment]
> My overall concern is the website loading speed. I'm not sure how you've done it, but your version is already loading faster than our current website. So kudos for that! But I think it's still pretty slow and should be improved (if possible).
Could we implement some kind of lazy loading? So the website is loading the first 100 (or 200, or 300, ...) icons and then if you scroll down (or click on a "Next Page" button), it will load the next 100? So we don't have to load all of the 1500+ SVGs at once.

This is also a concern of mine, but since it's a problem on the existing site already I didn't consider it for a minimum viable product. I did try to make the loading speed as fast of possible using some tricks, and have some other ideas I would like to apply to improve it even more. There actually already is an issue for lazy loading, #2384, the main difficulty with that is the two sorting methods we offer as well as making that work nicely with search. Moreover, lazy loading would additionally require some nice solution in case JavaScript is disabled (probably pagination, but not sure how to implement that without a backend...)


> I'm not too sure if I'm a big fan of the current way of displaying the same logo twice (once in black and once in light grey and half cut off). Either I would prefer to display the logo in light grey without cutting it off:

I can see why you think that, I'm not 100% on it myself either. I don't particularly like the designs you proposed as they are either, but I suppose they're just quick and dirty. I will try to play around with the design a bit to see if it can be improved.

> I guess by displaying (and therefore loading) the SVGs only once, we could as well improve the website loading speed significantly.

Actually, it would barely improve the loading speed as the "background" version is just [an `<svg>` tag that references the the copyable version](https://github.com/ericcornelissen/simple-icons-website/blob/61fb9d80c87889b77bc13c076876e738426d13de/index.pug#L78-L79).


> Another last suggestion: I think it would look better, if the cards in the last row would keep their size. Currently they are stretched out:

I agree, the redesign is purely based on flexbox, whereas the existing implementation uses a grid if the browser supports it. With grid it would be pretty easy to achieve that but haven't been able to get it working with Flexbox yet... Anyway, I definitely plan to fix this :+1:

**Comment by runxel** [comment]
Oops, sorry. Didn't want to close.

Great work, @ericcornelissen! That looks fantastic already! 💯 
Personally, I like the greyed out logo in the corner. Looks slick.


By the way: Any chance of having a dark mode? 😅 I could provide that one, too.

**Comment by ericcornelissen** [comment]
Thanks @runxel :smile:

> By the way: Any chance of having a dark mode? sweat_smile I could provide that one, too.

Was already on my list and actually started working on it today (before I saw your comment). You can have a sneak peak by checking out https://github.com/ericcornelissen/simple-icons-website/pull/5

**Comment by runxel** [comment]
Nice @ericcornelissen!
I really like the work you've done.

**Comment by el1f** [comment]
Looks great! Also a big fan of the light corner logo since it gives it a little visual flare with next to no cost.
Was wondering about why you decided to opt for a full flexbox implementation instead of CSS Grid though. Is it just because of browers compatibility?

**Comment by ericcornelissen** [comment]
> Looks great! Also a big fan of the light corner logo since it gives it a little visual flare with next to no cost.
> Was wondering about why you decided to opt for a full flexbox implementation instead of CSS Grid though. Is it just because of browers compatibility?

Primarily because of browser compatibility. The current website [specifies the layout using both flexbox and grid](https://github.com/simple-icons/simple-icons/blob/7affe87312bdeef4e60a492ecd6602f9f403a2ba/stylesheet.css), which seemed a bit redundant to me so I tried to achieve the layout with flexbox only. However, [as I said earlier](https://github.com/simple-icons/simple-icons/issues/980#issuecomment-752982604) it seems like we are going to require grid to achieve the nicest version of the layout, with flexbox as a fallback for older browsers (which may be removed in the future).

**Comment by el1f** [comment]
>Interestingly, the use grid significantly (~1 full second) increases load speed (due to page painting) so I actually think I will be reverting this change 🤔 at least for now...

Ain't this interesting! Definitely gonna keep in mind for future projects! Thanks for the info!

**Comment by ericcornelissen** [comment]
> Interestingly, the use grid significantly (~1 full second) increases load speed due to page painting.

I did some further testing and to clarify: I don't think this is something to consider for most websites as it seems to be caused by the combination of using [`minmax(14rem, 1fr)` on repeat](https://github.com/ericcornelissen/simple-icons-website/commit/34587554409f9c93ab7b77c01eea4fabb3351dd4#diff-d5e25bdbe5815ca8804129249aa126a217444fcdf8e45a04f2ef6e3da4b8daf6R404) with over 1500 DOM elements. Additionally, I did some proper measurements now and for me using grid is about ~800ms slower than using flexbox.

I'm now convinced that we should avoid using grid for now, at least until we look into https://github.com/simple-icons/simple-icons/issues/2384.

**Comment by ericcornelissen** [comment]
Quick update, I migrated the redesigned website repository to this organisation. You can now find it at:

- repo: https://github.com/simple-icons/simple-icons-website
- website: https://simple-icons.github.io/simple-icons-website/

**Comment by ericcornelissen** [comment]
Perhaps a little bit premature, but I already have [a branch ready to remove the website from this repository](https://github.com/ericcornelissen/simple-icons/tree/remove-website).

However, there is one thing that I'm unsure of and it is [a15d06972b233c59780a93be9dd0b342469f420e](https://github.com/ericcornelissen/simple-icons/commit/a15d06972b233c59780a93be9dd0b342469f420e). I have never used GitPod for this project and so am unsure how it is used in practice. Since I believe it is mainly used to preview new icons in the website, and the website is being removed, I removed the GitPod configuration as well. If anyone has objections to this I'm happy to add the GitPod configuration back.

**Comment by jorgeamadosoria** [comment]
@ericcornelissen are there any action items that you hadn't covered in other issues / PRs? I've read a few comments and they are apparently either covered or superseded by other changes.

Trying to gain clarity on which action items need to be separated in their own tickets to close this one, as it is too long and deals with many things at once.

**Comment by ericcornelissen** [comment]
I tried to create an issue in the website repo for everything in this thread that was not covered by the initial launch of the new website. But as the thread is pretty large I may have missed some things.
